src/bsoncxx/types.hpp

```diff
@@ -16,6 +16,8 @@
 
 #include <chrono>
 #include <cstring>
+#include <math.h>
```

Comment:
Using <cmath>, <climits>, etc. will ensure that C library symbols are in the std:: namespace-- especially since this is a header file that will be included by other programs.

Reply:
Ok, made the solution more proper C++ by including <cmath> and <limits> instead of .h versions, and using std::fabs explicitly.